expression/builtin_cast_vec.go

```diff
@@ -38,3 +39,40 @@ func (b *builtinCastIntAsIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Co
 func (b *builtinCastIntAsIntSig) vectorized() bool {
 	return true
 }
+
+func (b *builtinCastIntAsRealSig) vecEvalReal(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ResizeFloat64(n, false)
+
+	i64s := buf.Int64s()
+	rs := result.Float64s()
+
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.SetNull(i, true)
+			continue
+		}
+		if !mysql.HasUnsignedFlag(b.tp.Flag) && !mysql.HasUnsignedFlag(b.args[0].GetType().Flag) {
```

Comment:
How about using two variables to store the results of `mysql.HasUnsignedFlag(b.tp.Flag)` and `mysql.HasUnsignedFlag(b.args[0].GetType().Flag)` outside this loop so that we can decrease some function calls.

Reply:
quite appreciate, it is helpful.